checks/runchecks.go

```diff
@@ -94,13 +99,74 @@ func RunChecks(ctx *context.Context) ([]*pkg.CheckResult, error) {
 		result := c.Run(ctx)
 		transformedResults := TransformResults(ctx, result)
 		results = append(results, transformedResults...)
-
 		ExportCheckMetrics(ctx, transformedResults)
 	}
 
+	if err := saveArtifacts(ctx, results); err != nil {
+		logger.Errorf("error saving artifacts: %v", err)
+	}
+
 	return ProcessResults(ctx, results), nil
 }
 
+func saveArtifacts(ctx *context.Context, results pkg.Results) error {
+	if DefaultArtifactConnection == "" {
+		return nil
+	}
+
+	connection, err := ctx.HydrateConnectionByURL(DefaultArtifactConnection)
+	if err != nil {
+		return fmt.Errorf("error getting connection(%s): %w", DefaultArtifactConnection, err)
+	} else if connection == nil {
+		return fmt.Errorf("connection(%s) was not found", DefaultArtifactConnection)
+	}
+
+	fs, err := GetFSForConnection(utils.Ptr(ctx.Duty()), *connection)
+	if err != nil {
+		return fmt.Errorf("error getting filesystem for connection: %w", err)
+	}
+	defer fs.Close()
+
+	for _, r := range results {
+		if len(r.Artifacts) == 0 {
```

Comment:
We probably want to move this to the top of the function, of no artifacts have been returned don't even look for a connection